### PR TITLE
Fix overrides bug

### PR DESCRIPTION
**Pull request by reshmabidikar**
There is a bug in the code due to which it is currently not possible to override a usage price by itself, it is necessary to update either a fixed or a recurring price too. This PR attempts to address this bug.

See [this comment](https://github.com/killbill/killbill-aviate-plugin/pull/287#discussion_r1886239492).

### CODE REVIEW

**Comment by reshmabidikar** [comment]
If you agree that this is a bug, I can write a test for this fix.

**Comment by reshmabidikar** [comment]
For 2/- above, I've added a test  See c29ce39. 
For 1/ above, I was not sure where to throw an exception. So I've added a validation to check that there is at least one overridden price and a test for this. With this change, there is now no need for [this check](https://github.com/killbill/killbill/blob/d7ff936609bc54a5129c080e5c450b781786d11c/jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/SubscriptionResourceHelpers.java#L215). See e941bba. If this is not what you meant please let me know and I can make changes accordingly.

File: profiles/killbill/src/test/java/org/killbill/billing/jaxrs/TestEntitlement.java [hunk 1]
```diff
@@ -2049,4 +2053,61 @@ public Boolean call() throws Exception {
...
+                                           ExtBusEventType.SUBSCRIPTION_CREATION);
+
+        final Subscription entitlementJson = subscriptionApi.createSubscription(input,
+                                                                                null,
+                                                                                (LocalDate) null,
+                                                                                NULL_PLUGIN_PROPERTIES,
+                                                                                requestOptions);
+        callbackServlet.assertListenerStatus();
+        final Subscription subscription = subscriptionApi.getSubscription(entitlementJson.getSubscriptionId(), requestOptions);
+        Assert.assertEquals(subscription.getState(), EntitlementState.ACTIVE);
```
Comment (sbrossie) [inline profiles/killbill/src/test/java/org/killbill/billing/jaxrs/TestEntitlement.java:2145]: Any way we could check that price was correctly overridden - would `subscription#getLastActivePlan()` provide such info?

File: jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/SubscriptionResourceHelpers.java [hunk 2]
```diff
@@ -93,7 +93,8 @@ public static List<PlanPhasePriceOverride> buildPlanPhasePriceOverrides(final It
         if (priceOverrides != null) {
             for (final PhasePriceJson input : priceOverrides) {
                 Preconditions.checkNotNull(input);
-
+                Preconditions.checkArgument(input.getFixedPrice() != null || input.getRecurringPrice() != null || input.getUsagePrices() != null || (input.getUsagePrices() != null && !input.getUsagePrices().isEmpty()),
```
Comment (sbrossie) [inline jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/SubscriptionResourceHelpers.java:96]: I think we can remove the  `input.getUsagePrices() != null` and simply keep the `(input.getUsagePrices() != null && !input.getUsagePrices().isEmpty()`

File: jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/SubscriptionResourceHelpers.java [hunk 3]
```diff
@@ -212,8 +213,6 @@ public List<UsagePriceOverride> getUsagePriceOverrides() {
         final PlanPhaseSpecifier planPhaseSpecifier = spec.getPlanName() != null ?
                                                       new PlanPhaseSpecifier(spec.getPlanName(), phaseType) :
                                                       new PlanPhaseSpecifier(spec.getProductName(), spec.getBillingPeriod(), spec.getPriceListName(), phaseType);
-        final Currency resolvedCurrency = input.getFixedPrice() != null || input.getRecurringPrice() != null || (input.getUsagePrices() != null && !input.getUsagePrices().isEmpty()) ? currency : null;
```
Comment (sbrossie) [inline jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/SubscriptionResourceHelpers.java]: Good point, I don't think this is needed.

**Review by sbrossie — APPROVED** [review]

File: profiles/killbill/src/test/java/org/killbill/billing/jaxrs/TestEntitlement.java (hunk 1 above)
Comment (reshmabidikar) [reply profiles/killbill/src/test/java/org/killbill/billing/jaxrs/TestEntitlement.java:2145]: Addressed in 984a82d

File: jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/SubscriptionResourceHelpers.java (hunk 2 above)
Comment (reshmabidikar) [reply jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/SubscriptionResourceHelpers.java]: Addressed in 984a82d

**Pull request merged**